common/logger.go

```diff
@@ -0,0 +1,24 @@
+// The common package holds types and variables that are common to
+// multiple parts of Felice, and generically useful.
+package common
```

Comment:
Dave Cheney doesn't agree with your package name!
- https://twitter.com/davecheney/status/697551191752855552?lang=fr
- https://dave.cheney.net/2016/08/20/solid-go-design

More seriously, I think we can simplify the way we log. The point of having a logger in the library is for debugging purpose, users usually don't need to format that in a specific way (other than what was decided by the library owner I mean), all that's important is where these logs are sent.
So I'm not sure we should provide a package or an interface for that.

I think at this stage we can simply have a `Logger *log.Logger` field in the `consumer.Consumer` type, it will allow us to:
- provide an API for the user to change the logger: `c.Logger = log.New(os.Stderr)`
- log to `ioutil.Discard` by default
- avoid using a global variable: https://dave.cheney.net/2017/01/23/the-package-level-logger-anti-pattern
- have a separate logger for each component of the system

Also, if a user absolutely needs to change the way logs are formatted, he can use a custom io.Writer, `zerolog` does that:

```go
log := zerolog.New(os.Stdout).With().
    Str("foo", "bar").
    Logger()

stdlog.SetFlags(0)
stdlog.SetOutput(log)
```

Reply:
TIL

Reply:
Done